notebooks/week1/hw1.jl

```diff
@@ -0,0 +1,1418 @@
+    
+Here, `::AbstractRGB` is a **type annotation**. This ensures that this version of the function will be chosen when passing in an object whose type is a **subtype** of the `AbstractRGB` abstract type. For example, both the `RGB` and `RGBX` types satisfy this.
+
+The method you write should return a new `RGB` object, in which each component ($r$, $g$ and $b$) are quantized. Use your previous method for `quantize`!
+"""
+
+# ╔═╡ 04e6b486-ceb7-45fe-a6ca-733703f16357
+function quantize(color::AbstractRGB)
+	# your code here!
+	return missing
+end
+
+# ╔═╡ f6bf64da-ee07-11ea-3efb-05af01b14f67
+md"""
+#### Exercise 3.4
+👉 Write a method `quantize(image::AbstractMatrix)` that quantizes an image by quantizing each pixel in the image. (You may assume that the matrix is a matrix of color objects.)
+"""
+
+# ╔═╡ 13e9ec8d-f615-4833-b1cf-0153010ccb65
+function quantize(image::AbstractMatrix)
+	# your code here!
+	return missing
+end
+
+# ╔═╡ f6a655f8-ee07-11ea-13b6-43ca404ddfc7
+quantize(0.267), quantize(0.91)
+
+# ╔═╡ 25dad7ce-ee0b-11ea-3e20-5f3019dd7fa3
+md"Let's apply your method!"
+
+# ╔═╡ 9751586e-ee0c-11ea-0cbb-b7eda92977c9
+quantize(philip)
+
+# ╔═╡ f6d6c71a-ee07-11ea-2b63-d759af80707b
+md"""
+#### Exercise 3.5
+👉 Write a function `noisify(x::Number, s)` to add randomness of intensity $s$ to a value $x$, i.e. to add a random value between $-s$ and $+s$ to $x$. If the result falls outside the range $[0, 1]$ you should "clamp" it to that range. (Julia has a built-in `clamp` function, or you can write your own function.)
+"""
+
+# ╔═╡ f38b198d-39cf-456f-a841-1ba08f206010
+function noisify(x::Number, s)
+	# your code here!
+	return missing
+end
+
+# ╔═╡ f6fc1312-ee07-11ea-39a0-299b67aee3d8
+md"""
+👉  Write the second method `noisify(c::AbstractRGB, s)` to add random noise of intensity $s$ to each of the $(r, g, b)$ values in a colour. 
+
+Use your previous method for `noisify`. _(Remember that Julia chooses which method to use based on the input arguments. So to call the method from the previous exercise, the first argument should be a number.)_
+"""
+
+# ╔═╡ db4bad9f-df1c-4640-bb34-dd2fe9bdce18
+function noisify(color::AbstractRGB, s)
+	# your code here!
+	return missing
+end
+
+# ╔═╡ 0000b7f8-4c43-4dd8-8665-0dfe59e74c0a
+md"""
+Noise strength:
+"""
+
+# ╔═╡ 774b4ce6-ee1b-11ea-2b48-e38ee25fc89b
+@bind color_noise Slider(0:0.01:1, show_value=true)
+
+# ╔═╡ f70823d2-ee07-11ea-2bb3-01425212aaf9
+md"""
+👉 Write the third method `noisify(image::AbstractMatrix, s)` to noisify each pixel of an image. This function should be a single line!
+"""
+
+# ╔═╡ 21a5885d-00ab-428b-96c3-c28c98c4ca6d
+function noisify(image::AbstractMatrix, s)
+	# your code here!
+	return missing
+end
+
+# ╔═╡ 1ea53f41-b791-40e2-a0f8-04e13d856829
+noisify(0.5, 0.1) # edit this test case!
+
+# ╔═╡ 7e4aeb70-ee1b-11ea-100f-1952ba66f80f
+(original=color_red, with_noise=noisify(color_red, color_noise))
+
+# ╔═╡ 8e848279-1b3e-4f32-8c0c-45693d12de96
+[
+	noisify(color_red, strength)
+	for 
```

Comment:
We can write and MD cell or a comment here if you want them to start thinking about list comprehensions now to just note how you are doing this.